### ISSUE TITLE
PARSEC implementations

### ISSUE THREAD

**Issue by jurrewolff** [libheartbeats/heartbeats#4]
Hi,

For a master thesis we'll most likely be using the heartbeat framework. The idea is to use the heartbeats in combination with the HotSniper toolchain, which runs PARSEC programs on simulated processor chips.

In the paper covering the first version of the heartbeat framework, several parsec programs were modified to include heartbeat monitoring. The source of this doesn't seem to be available online anymore unfortunately.

Is there an implementation of heartbeats 2.0 in PARSEC programs available? If not, do you still have the implementations of the first version stored somewhere?

Thank you in advance.

Regards,
Jurre

**Comment by connorimes** [comment]
Hi Jurre,

As part of the POET and Bard projects, I made patches available on the projects' website [1], but note that these contain both the addition of Heartbeats 2.0 and the POET or Bard tools.  At some later time I made my own PARSEC repository public [2], with `poet` and `bard` branches that I generated the patches from.  I believe the original PARSEC source I used is still available [3].

I don't think I have a modified PARSEC version that only includes Heartbeats 2.0, but it shouldn't be hard to just strip out the POET or Bard stuff or use the patches as a reference to modify the upstream source yourself.  My recollection is that harder part is just getting PARSEC to compile at all with modern versions of gcc.

[1] http://poet.cs.uchicago.edu
[2] https://github.com/connorimes/parsec-3.0
[3] https://parsec.cs.princeton.edu/parsec3-doc.htm

Cheers.
